taurus/util/impl.py

```diff
@@ -57,7 +62,7 @@ def _substitute(thing,
     elif isinstance(thing, DictSerializable):
         new_attrs = {_substitute(k, sub, applies, visited): _substitute(v, sub, applies, visited)
                      for k, v in thing.as_dict().items()}
-        new = thing.from_dict(new_attrs)
+        new = thing.build(new_attrs)
```

Comment:
This is what fixes https://github.com/CitrineInformatics/citrine-python/pull/203 as far as I can tell